src/core/drive/page_renderer.ts

```diff
@@ -127,42 +113,42 @@ export class SnapshotRenderer extends Renderer {
   }
 
   assignNewBody() {
-    if (document.body) {
-      replaceElementWithElement(document.body, this.newBody)
+    if (document.body && this.newElement instanceof HTMLBodyElement) {
+      replaceElementWithElement(document.body, this.newElement)
     } else {
-      document.documentElement.appendChild(this.newBody)
+      document.documentElement.appendChild(this.newElement)
     }
   }
 
   focusFirstAutofocusableElement() {
-    const element = this.newSnapshot.findFirstAutofocusableElement()
+    const element = this.newSnapshot.firstAutofocusableElement
     if (elementIsFocusable(element)) {
       element.focus()
     }
   }
 
   getNewHeadStylesheetElements() {
```

Comment:
I believe these are private functions, so maybe the payoff is limited, but is now a good opportunity to re-structure these as properties?:

```js
get newHeadStylesheetElements() {
  // ...
}
```

Reply:
Absolutely, yeah, these should be getters.